subprojects/gradle/lint-report/src/main/kotlin/com/avito/android/lint/slack/LintSlackReporter.kt

```diff
@@ -25,43 +27,112 @@ interface LintSlackReporter {
         override fun report(
             lintReport: LintReportModel,
             channel: SlackChannel,
+            channelForLintBugs: SlackChannel,
             buildUrl: HttpUrl
         ) {
-            if (shouldSendAlert(lintReport)) {
-                logger.debug("$tag: Sending lint alert...")
-
-                slackClient.uploadHtml(
-                    channel = channel,
-                    message = buildSlackMessage(lintReport, buildUrl),
-                    file = lintReport.htmlFile
-                ).fold(
-                    { logger.debug("$tag: Report sent successfully") },
-                    { error -> logger.critical("$tag: Can't send report ${error.message}") }
-                )
-            } else {
-                logger.debug("$tag Skip sending lint alert")
+            when (lintReport) {
+                is LintReportModel.Valid -> {
+                    val (errors, everythingElse) = lintReport.issues.partition { it.severity == LintIssue.Severity.ERROR }
+                    val (warnings, unknowns) = everythingElse.partition { it.severity == LintIssue.Severity.WARNING }
+                    val (fatalErrors, regularErrors) = errors.partition { it.isFatal }
+
+                    var isMessageSent = false
+
+                    if (regularErrors.isNotEmpty()) {
+                        sendReport(
+                            channel = channel,
+                            message = buildSlackMessage(
+                                projectPath = lintReport.projectRelativePath,
+                                errors = regularErrors,
+                                warnings = warnings,
+                                buildUrl = buildUrl
+                            ),
+                            htmlReport = lintReport.htmlFile
+                        )
+
+                        isMessageSent = true
+                    }
+
+                    if (fatalErrors.isNotEmpty() || unknowns.isNotEmpty()) {
+                        sendReport(
+                            channel = channelForLintBugs,
+                            message = buildSlackMessageAboutLintBugs(
+                                projectPath = lintReport.projectRelativePath,
+                                fatalErrors = fatalErrors,
+                                unknownErrors = unknowns,
+                                buildUrl = buildUrl
+                            ),
+                            htmlReport = lintReport.htmlFile
+                        )
+
+                        isMessageSent = true
```

Comment:
Why do we need this state?
Can we avoid it or store it somewhere else (task output, check by API in slack, ...)?